tests/test_project.py

```diff
@@ -2317,6 +2318,236 @@ def test_main_submit(self, monkeypatch):
+        "installed_success",
+        "installed_fail",
+    ]
+
+    @pytest.fixture(params=["base_cmd", "base_cmd_no_decorators"])
+    def operation_name(self, request):
+        return request.param
+
+
+class TestHooksInstallWithDecorators(TestHooksBase, TestHooksInstallSetUp):
+    # Tests if project-wide hooks interfere with operation level hooks
+    @pytest.fixture(params=["base"])
+    def operation_name(self, request):
+        return request.param
+
+
+class TestHooksInstallCmdWithDecorators(TestHooksCmd, TestHooksInstallSetUp):
+    # Tests if project-wide hooks interfere with operation level hooks
+    # in job operations with the @cmd decorator
+    @pytest.fixture()
+    def operation_name(self):
+        return "base_cmd"
+
+
+class TestHooksInstallNoDecorators(TestHooksInstallSetUp):
+    # Tests if operation level hooks interfere with project-level hooks
+    @pytest.fixture(params=["base_no_decorators", "base_cmd_no_decorators"])
+    def operation_name(self, request):
+        return request.param
+
+    @pytest.fixture()
+    def job(self, project):
+        return project.open_job(dict(raise_exception=False))
+
+    def test_no_decorator_keys(self, operation_name, job):
+        get_job_doc_key = self._get_job_doc_key(job, operation_name)
+        self.call_subcmd(f"run -o {operation_name} -j {job.id}")
+        for key in self.keys:
+            assert get_job_doc_key(key) is None
+
+
+class TestHooksInvalidOption(TestHooksSetUp):
+    def call_subcmd(self, subcmd, stderr=subprocess.STDOUT):
+        # Return error as output instead of raising error
+        fn_script = self.entrypoint["path"]
+        _cmd = f"python {fn_script} {subcmd} --debug"
+        with add_path_to_environment_pythonpath(os.path.abspath(self.cwd)):
+            try:
+                with switch_to_directory(self.project.root_directory()):
+                    return subprocess.check_output(_cmd.split(), stderr=stderr)
+            except subprocess.CalledProcessError as error:
+                return str(error.output)
+
+    def test_invalid_hook(self):
+        class A(FlowProject):
+            pass
+
+        with pytest.raises(AttributeError):
+
+            @A.operation
+            @A.add_hook.invalid_option(lambda operation_name, job: None)
+            def test_invalid_decorators(_):
+                pass
+
+    def test_install_invalid_hook(self):
+        class InstallInvalidHook:
+            def install_hook(self, project):
+                project.hooks.invalid_option.append(lambda operation_name, job: None)
+
+        with pytest.raises(AttributeError):
+            InstallInvalidHook().install_hook(self.mock_project())
+
+    def test_raise_exception_in_hook(self):
+        job = self.mock_project().open_job(dict(raise_exception=False))
+
+        error_output = self.call_subcmd(f"run -o raise_exception_in_hook -j {job.id}")
+
+        assert "RuntimeError" in error_output
+        assert "Error occurred during execution of "
```

Comment:
What is this supposed to assert?

Reply:
I removed it. I think the original intention was to make sure the error messages matched up and I didn't catch the wrong error.